xray/client/xray_viewer.ts

```diff
@@ -98,23 +107,22 @@ export class XRayViewer {
       edge_length_px_for_level *= 2;
       level += 1;
     }
+    this.displayLevel = level;
 
     // We encode the matrix column major.
-    const request = new Request(
-      `${this.prefix}/nodes_for_level?level=${level}&matrix=${matrixToString(
-        matrix
-      )}`,
-      {
-        method: 'GET',
-        credentials: 'same-origin',
-      }
-    );
-
+    this.currentForLevelQueryNumber++;
+    let queryNumber = this.currentForLevelQueryNumber;
     window
-      .fetch(request)
+      .fetch(
+        `${this.prefix}/nodes_for_level?level=${level}&matrix=${matrixToString(
+          matrix
+        )}`
+      )
       .then((data) => data.json())
       .then((nodes: any) => {
-        this.nodesUpdate(nodes);
+        if (this.currentForLevelQueryNumber == queryNumber) {
```

Comment:
I would add a comment like "Ignore responses that arrive after we've already issued a new request."

Not for this PR, but you might also want to look into using [axios](https://www.npmjs.com/package/axios), which (unlike `fetch`) allows canceling requests.

Reply:
done. Added a TODO for axios.